manubot/cite/util.py

```diff
@@ -36,21 +36,38 @@
 
 def standardize_citation(citation):
     """
-    Standardize citation idenfiers based on their source
+    Standardize citation identifiers based on their source
     """
     source, identifier = citation.split(':', 1)
+
     if source == 'doi':
+        if identifier.startswith('10/'):
+            from manubot.cite.doi import expand_short_doi
+            try:
+                identifier = expand_short_doi(identifier)
+            except Exception as error:
+                # If DOI shortening fails, return the unshortened DOI.
+                # DOI metadata lookup will eventually fail somewhere with
+                # appropriate error handling, as opposed to here.
+                logging.error(
```

Comment:
This error message prints twice if I try using a shortDOI that is invalid but passes the regex.  Is that expected?

`manubot cite doi:10/98p1`

Reply:
That is because `standardize_citation` was being called by `citation_to_citeproc` to assert that the input citation is already standardized:

https://github.com/manubot/manubot/blob/1b3ac66de9220e0b7496128a4a45e444b6d4edb1/manubot/cite/util.py#L216-L220

I am not sure whether we really need to do that check, but the easiest solution was to cache `standardize_citation` in https://github.com/manubot/manubot/pull/93/commits/8ed947318bf483769de78b5ec490337e87d23221. This should have potential performance improvements as well.